mesos-rxjava-client/src/main/java/com/mesosphere/mesos/rx/java/MesosClientBuilder.java

```diff
@@ -163,6 +166,108 @@ private MesosClientBuilder() {}
+     * will be invoked, and the overflow will mitigate the issue based on the {@link BackpressureOverflow.Strategy}
+     * that you select.
+     *
+     * <ul>
+     *     <li>{@code BackpressureOverflow.Strategy.ON_OVERFLOW_ERROR} (default) will {@code onError} dropping all undelivered items,
+     *     unsubscribing from the source, and notifying the producer with {@code onOverflow}. </li>
+     *     <li>{@code BackpressureOverflow.Strategy.ON_OVERFLOW_DROP_LATEST} will drop any new items emitted by the producer while
+     *     the buffer is full, without generating any {@code onError}.  Each drop will however invoke {@code onOverflow}
+     *     to signal the overflow to the producer.</li>
+     *     <li>{@code BackpressureOverflow.Strategy.ON_OVERFLOW_DROP_OLDEST} will drop the oldest items in the buffer in order to make
+     *     room for newly emitted ones. Overflow will not generate an{@code onError}, but each drop will invoke
+     *     {@code onOverflow} to signal the overflow to the producer.</li>
+     * </ul>
+     *
+     * As an example, this may be necessary for Mesos schedulers that launch large numbers
+     * of tasks at a time and then request reconciliation.
+     *
+     * @param capacity number of slots available in the buffer.
+     * @param onOverflow action to execute if an item needs to be buffered, but there are no available slots.  Null is allowed.
+     * @param strategy how should the {@code Observable} react to buffer overflows.  Null is not allowed.
+     * @return this builder (allowing for further chained calls)
+     * @see <a href="http://reactivex.io/documentation/operators/backpressure.html">ReactiveX operators documentation: backpressure operators</a>
+     */
+    @NotNull
+    public MesosClientBuilder<Send, Receive> onBackpressureBuffer(
+            @NotNull final long capacity,
+            @NotNull final Action0 onOverflow,
+            @NotNull final BackpressureOverflow.Strategy strategy
+            ) {
+        this.byteStreamTransformer = observable -> observable.onBackpressureBuffer(capacity, onOverflow, strategy);
+        return this;
+    }
+
+    /**
+     * <b>
+     *     WARN: This is intended for advanced usage only. Use with caution.
+     * </b>
+     *
+     * This allows an arbitrary {@link rx.Observable.Transformer} to be placed in the Mesos HTTP byte[] stream
+     * between the {@link com.mesosphere.mesos.rx.java.recordio.RecordIOOperator} and the
+     * {@link rx.internal.operators.OperatorObserveOn} using {@code compose}
+     *
+     * Calling {@code builder.onBackpressureBuffer()} is equivalent to calling
+     * {@code builder.transformByteStream(obs -> obs.onBackpressureBuffer()}
+     *
+     *
+     * @param transformer Arbitrary transformer of one byte[] observable to another
+     * @return this builder (allowing for further chained calls)
+     */
+    @NotNull
+    public MesosClientBuilder<Send, Receive> transformByteStream(
```

Comment:
There isn't anything I can think of right now, and would prefer to keep the api surface area as small as possible, so let's go ahead and remove this method.